V0.1.1 Adding pytest for bigquery

tusharchou opened the pull request:
<img width="1345" alt="Screenshot 2024-11-04 at 7 25 18 PM" src="https://github.com/user-attachments/assets/29c11e27-1611-4857-8b51-be6dae9fcc3c">


---
Name: pytest for bigquery
About: Implement a test for class source bigquery
title: ''
labels: ''
assignees: ''

---

**Is your pull request related to a issue? Please tag.**

**Describe the solution you have implemented**
Describe the solution you have implemented

I implemented a test for the source_bigquery class using pytest. This includes covering various scenarios to ensure the class behaves as expected with different inputs and configurations.

Describe why alternatives you considered are not optimal

Using unittest was considered, but pytest was chosen for its concise and readable syntax, making the tests easier to maintain and understand.

Additional context

The tests were conducted against a mock BigQuery setup to ensure they can be executed in a local environment without needing access to an actual BigQuery instance.

Here's a screenshot of the test results:

tusharchou committed:
V0.1.1 (#58)

tusharchou committed:
0.1.1 BigQuery Release  (#46)

* Create publish.yml

* Update publish.yml

tusharchou committed:
pypi release through github actions (#60)

* Create publish.yml

* Update publish.yml

* Update publish.yml

tusharchou committed:
Create manual.yml

tusharchou committed:
Update publish.yml (#62)

tusharchou committed:
Update publish.yml (#63)

tusharchou committed:
Update pyproject.toml (#64)

tusharchou committed:
Update publish.yml (#65)

* Update publish.yml

* Update publish.yml

* Update publish.yml

brmhastra committed:
Brmhastra patch 1 (#67)

* Update publish.yml

* Update publish.yml

* Update publish.yml

* Update publish.yml

updated yaml file to copy distribution o/p from build to root directory

* Update publish.yml

added detailed copy from /ldf/dist to gihub/wo../dist

---------

Co-authored-by: Tushar Choudhary <151359025+tusharchou@users.noreply.github.com>

brmhastra committed:
Brmhastra patch 1 (#68)

* Update publish.yml

* Update publish.yml

* Update publish.yml

* Update publish.yml

updated yaml file to copy distribution o/p from build to root directory

* Update publish.yml

added detailed copy from /ldf/dist to gihub/wo../dist

* Update publish.yml

added a code for creating dist directory

---------

Co-authored-by: Tushar Choudhary <151359025+tusharchou@users.noreply.github.com>

brmhastra committed:
Brmhastra patch 1 (#69)

* Update publish.yml

* Update publish.yml

* Update publish.yml

* Update publish.yml

updated yaml file to copy distribution o/p from build to root directory

* Update publish.yml

added detailed copy from /ldf/dist to gihub/wo../dist

* Update publish.yml

added a code for creating dist directory

* Update publish.yml

relocating dots from ./github to /.github

---------

Co-authored-by: Tushar Choudhary <151359025+tusharchou@users.noreply.github.com>

mrutunjay-kinagi committed:
GitHub workflow for pypi release (#70)

* Update publish.yml

* Update pyproject.toml

mrutunjay-kinagi committed:
GitHub workflow for pypi release (#71)

* Update publish.yml

* Update pyproject.toml

* Update publish.yml

mrutunjay-kinagi committed:
Update publish.yml (#72)

mrutunjay-kinagi committed:
Update publish.yml (#73)

mrutunjay-kinagi committed:
Release v1.1 changes (#74)

mrutunjay-kinagi committed:
Github action setup for release v1.1 (#76)

* Release v1.1 dist changes

* Release v1.1 publish.yml changes

* Release v1.1 publish.yml changes

redpheonixx committed:
Update publish.yml

redpheonixx committed:
Merge pull request #77 from tusharchou/redpheonixx-patch-1

Update publish.yml

redpheonixx committed:
Merge pull request #61 from tusharchou/tusharchou-patch-4

Create manual.yml

redpheonixx committed:
Update publish.yml (#78)

redpheonixx committed:
Update publish.yml

redpheonixx committed:
Merge pull request #79 from tusharchou/redpheonixx-patch-3

Update publish.yml

redpheonixx committed:
Update publish.yml (#80)

redpheonixx committed:
Update pyproject.toml (#81)

tusharchou reviewed (commented):
todo
- add following to git ignore
- .whl 
- .zip
- .gz
-

mrutunjay-kinagi committed:
Release v1.1 fix (#84)

* Release v1.1 dist changes

* Release v1.1 publish.yml changes

* Release v1.1 publish.yml changes

* Release v1.1

mrutunjay-kinagi commented:
@tusharchou  Please resolve this conflict. 
Run the following commands : 
1. git fetch origin 
2. git switch main
3. git pull 
4. git switch <branch_with_conflict>
5. git merge main
6. Resolve conflicts and commit

redpheonixx commented on local-data-platform/local_data_platform/__init__.py:
```diff
@@ -39,7 +39,7 @@ class Table(Base):
     def __init__(self, name: str, path: Path = os.getcwd()):
         self.name = name
-        self.path = path
+        self.path = os.getcwd()+path
```
why you have taken os.getcwd() twice, once as argument and other time concatenating ?
pls check

mrutunjay-kinagi committed:
0.1.1 Pytest Added for BigQuery Source  (#86)

* fixed bug warehouse uri

Thu Oct 24 9:25 PM IST

* added logging to CSV.get()

* supported big query ts format

* refactored parameter to config from catalog to reduce confusion

* Fixed bug of logger in GCP

* replaced local path with dynamic path

* replaced local path with dynamic path

* replaced local path with dynamic path

* demo

* demo

* Release v1.1 dist changes

* Release v1.1 publish.yml changes

* Release v1.1 publish.yml changes

* added a class implementation for github issue hoping it is useful for guiding users to resolution ETAs eventually

* added a exception for PlanNotFound to ask users to raise issues on the repository for resolution

* Updated overview and milestones. Added directory structure under technical specifications.

* Updated components in technical speciifcations

* added testing for BigQueryToCSV.extract()

* added testing for BigQueryToCSV.extract()

* added testing for Iceberg.get()

* Release v1.1

* Release v1.1 bug fix

* Release v1.1 bug fix

* Release v1.1 bug fix

* Pytest Added for BigQuery Source

---------

Co-authored-by: Tushar Choudhary <151359025+tusharchou@users.noreply.github.com>

tusharchou committed:
fix(deps): Stabilize build dependencies and configuration

redpheonixx committed:
Merge pull request #95 from tusharchou/fix/stabilize-dependencies

0.1.1 fix(deps): introducing mkdocs

sankalpmodi94 committed:
Add restaurant_data_mart_PRD.md for managing restaurant data with LDP (#93)

tusharchou committed:
feat(dx): Add Makefile and setup guide (#96)

* feat(dx): Add Makefile and setup guide

* feat: Stabilize build and enhance developer experience

- Fixes dependency issues by correcting pyproject.toml and standardizing on a root requirements.txt for Read the Docs.
- Adds a Makefile to streamline common development commands.
- Introduces Dockerfiles for reproducible development and documentation hosting.
- Implements a script to dynamically generate a list of open GitHub issues for the documentation.
- Cleans up the project structure by removing obsolete Sphinx files, duplicate Makefiles, and misplaced source files.

* Add docs improvements, recipes, test, and CI workflow

* Remove site/ assets from version control and add to .gitignore

* docs: add wiki to docs and update mkdocs navigation

* Refactor: move package to root, update docs, fix navigation and developer tooling

* cleaning

* fix: update readthedocs.yml and docs improvements

* deleted readthedocs.yml

* added more docs

* added more docs

* added letter

* feat(library): introduce github module and src layout

* reset

* issues

* issues fix

* homepage

* business solutions

* marketing data analysis

* marketing data analysis fix

* marketing data analysis final
